server/src/main/java/de/uftos/services/StudentGroupService.java

```diff
@@ -0,0 +1,65 @@
+package de.uftos.services;
+
+import de.uftos.dto.StudentsGroups;
+import de.uftos.entities.Student;
+import de.uftos.entities.StudentGroup;
+import de.uftos.repositories.StudentGroupRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.List;
+
+@Service
+public class StudentGroupService {
+    private final StudentGroupRepository repository;
+
+    @Autowired
+    public StudentGroupService(StudentGroupRepository repository) {
+        this.repository = repository;
+    }
+
+    public Page<StudentGroup> get(Pageable pageable) {
+        return this.repository.findAll(pageable);
+    }
+
+    public StudentGroup getById(String id) {
+        var group = this.repository.findById(id);
+
+        return group.orElseThrow(() -> new ResponseStatusException(HttpStatus.BAD_REQUEST));
+    }
+
+    public StudentGroup create(StudentGroup group) {
+        if (group.getId() != null && this.repository.findById(group.getId()).isPresent()) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST);
+        }
+
+        return this.repository.save(group);
+    }
+
+    public StudentGroup update(String id, StudentGroup group) {
+        group.setId(id);
+
+        return this.repository.save(group);
+    }
+
+    public void addStudents(String id, List<Student> students) {
+        this.repository.addStudentsToGroups(students.stream().map((student) -> new StudentsGroups(id, student.getId())).toList());
```

Comment:
Arguments of StudentsGroups constructor call switched.

Reply:
Good call!